lib/charon/session_store/redis_store.ex

```diff
@@ -46,101 +46,109 @@ defmodule Charon.SessionStore.RedisStore do
   @impl true
   def get(session_id, user_id, type, config) do
     mod_conf = get_mod_config(config)
-    {user_id_str, type_str, key_prefix} = key_data(user_id, type, mod_conf)
-    old_session_key = old_session_key(session_id, user_id_str, type_str, key_prefix)
-    session_key = session_key(session_id, user_id_str, type_str, key_prefix)
-
-    # TODO: all of this can be replaced by GET single key after a while
-    ["MGET", session_key, old_session_key]
-    |> mod_conf.redix_module.command()
+    key_data = key_data(user_id, type, mod_conf)
+    session_set_key = session_set_key(key_data)
+    oldest_session_key = oldest_session_key(session_id, key_data)
+    old_session_key = old_session_key(session_id, key_data)
+
+    [
+      ["HGET", session_set_key, session_id],
+      ["MGET", old_session_key, oldest_session_key]
+    ]
+    |> mod_conf.redix_module.pipeline()
     |> case do
-      {:ok, results} -> Enum.find_value(results, &deserialize(&1, mod_conf, config))
-      error -> error
+      {:ok, [h_result, m_results]} ->
+        Enum.find_value([h_result | m_results], &deserialize(&1, mod_conf, config))
+        |> validate(session_id, user_id, type, Internal.now())
```

Comment:
This grants precedence to newest storage format (hashset-based), then older (unhashed session key) then oldest (hashed session key)